enterprise/server/remote_execution/execution_server/execution_server.go

```diff
@@ -337,52 +337,53 @@ func (s *ExecutionServer) updateExecution(ctx context.Context, executionID strin
 }
 
 func (s *ExecutionServer) recordExecution(ctx context.Context, executionID string) error {
-	if s.env.GetExecutionCollector() == nil || !olapdbconfig.WriteExecutionsToOLAPDBEnabled() {
+	collector := s.env.GetExecutionCollector()
+	if collector == nil || !olapdbconfig.WriteExecutionsToOLAPDBEnabled() {
 		return nil
 	}
 	var executionPrimaryDB tables.Execution
-
 	if err := s.env.GetDBHandle().NewQuery(ctx, "execution_server_lookup_execution").Raw(
 		`SELECT * FROM "Executions" WHERE execution_id = ?`, executionID).Take(&executionPrimaryDB); err != nil {
 		return status.InternalErrorf("failed to look up execution %q: %s", executionID, err)
 	}
-	// Always clean up invocationLinks in Collector because we are not retrying
 	defer func() {
-		err := s.env.GetExecutionCollector().DeleteInvocationLinks(ctx, executionID)
-		if err != nil {
+		// Always clean up invocationLinks in Collector because we are not retrying
+		if err := collector.DeleteInvocationLinks(ctx, executionID); err != nil {
 			log.CtxErrorf(ctx, "failed to clean up invocation links in collector: %s", err)
 		}
 	}()
-	links, err := s.env.GetExecutionCollector().GetInvocationLinks(ctx, executionID)
 
+	links, err := collector.GetInvocationLinks(ctx, executionID)
 	if err != nil {
 		return status.InternalErrorf("failed to get invocations for execution %q: %s", executionID, err)
 	}
-
+	rmd := bazel_request.GetRequestMetadata(ctx)
 	for _, link := range links {
 		executionProto := execution.TableExecToProto(&executionPrimaryDB, link)
-		inv, err := s.env.GetExecutionCollector().GetInvocation(ctx, link.GetInvocationId())
+		if rmd != nil {
+			executionProto.TargetLabel = rmd.GetTargetId()
+		}
```

Comment:
seems like these 2 lines are the only changes needed? (maybe revert the other changes?)

Reply:
Yeah, I thought I would leave the cleanup code in as they are a bit nicer to look at.

I reverted them all. Let's keep the scope of change small.